src/v8/v8.cc

```diff
@@ -263,9 +268,41 @@ bool V8::load(const std::string &code, bool allow_precompiled) {
     assert((shared_module_ != nullptr));
   }
 
+  buildFunctionNameIndex();
+
   return module_ != nullptr;
 }
 
+void V8::buildFunctionNameIndex() {
+  // build function index -> function name map for backtrace
+  // https://webassembly.github.io/spec/core/appendix/custom.html#binary-namesubsection
+  auto name_section = getCustomSection("name");
+  if (name_section.size()) {
+    const byte_t *pos = name_section.data();
+    const byte_t *end = name_section.data() + name_section.size();
+    while (pos < end) {
+      if (*pos++ != 1) {
+        pos += parseVarint(pos, end);
+      } else {
+        auto size = parseVarint(pos, end);
+        auto start = pos;
+        const uint32_t namemap_vector_size = parseVarint(pos, end);
+        for (auto i = 0; i < namemap_vector_size; i++) {
+          const uint32_t func_index = parseVarint(pos, end);
+          const uint32_t func_name_size = parseVarint(pos, end);
+          function_names_index_.insert({func_index, std::string(pos, func_name_size)});
+          pos += func_name_size;
+        }
+
+        if (start + size != pos) {
+          // indicates the malformed function name subsection, so clear the stored indexes.
+          function_names_index_ = {};
+        }
```

Comment:
Nit: if this fails then we should probably `break` from the loop, since we can't trust the `pos` anyway... But I just realized that we don't do this type of verification when parsing bytecode in other places, so feel free to drop this whole `if` block if you prefer.

Reply:
yeah, we trust given binaries anywhere while parsing so far :smile: maybe we should decide and stick to whether or not validate binary or not. 

will add break for now.